developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -0,0 +1,283 @@
+
+import ManageDataCode from '!!raw-loader!/_includes/code/howto/manage-data.read.py';
+import ManageDataCodeV3 from '!!raw-loader!/_includes/code/howto/manage-data.read-v3.py';
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python (v4)">
+    <FilteredTextBlock
+      text={ManageDataCode}
+      startMarker="# ReadObject START"
+      endMarker="# ReadObject END"
+      language="py"
+    />
+  </TabItem>
+
+  <TabItem value="py3" label="Python (v3)">
+    <FilteredTextBlock
+      text={ManageDataCodeV3}
+      startMarker="# ReadObject START"
+      endMarker="# ReadObject END"
+      language="py"
+    />
+  </TabItem>
+</Tabs>
+
+Note here that the collection object can be re-used throughout the codebase.
+
+### Collection creation from JSON
+
+You can still create a collection from a JSON definition. This may be a useful way to migrate your existing data, for example. You could [fetch an existing definition](../../manage-data/collections.mdx#read-a-single-collection-definition) and then use it to create a new collection.
+
+<FilteredTextBlock
+  text={PythonCode}
+  startMarker="# START CreateCollectionFromJSON"
+  endMarker="# END CreateCollectionFromJSON"
+  language="py"
+/>
+
+### Removal of builder patterns
+
+The builder patterns for constructing queries as been removed, as they could be confusing and potentially lead to invalid queries.
+
+In `v4`, queries are constructed using specific methods and its parameters.
+
+import SearchSimilarityCode from '!!raw-loader!/_includes/code/howto/search.similarity.py';
+import SearchSimilarityCodeV3 from '!!raw-loader!/_includes/code/howto/search.similarity-v3.py';
+
+<Tabs groupId="languages">
+  <TabItem value="py" label="Python (v4)">
+    <FilteredTextBlock
+      text={SearchSimilarityCode}
+      startMarker="# GetNearTextPython"
+      endMarker="# END GetNearTextPython"
+      language="python"
+    />
+  </TabItem>
+
+  <TabItem value="py3" label="Python (v3)">
+    <FilteredTextBlock
+      text={SearchSimilarityCodeV3}
+      startMarker="# GetNearTextPython"
+      endMarker="# END GetNearTextPython"
+      language="python"
+    />
+  </TabItem>
+</Tabs>
+
+This makes it easier to understand and use. Additionally, some parameters typed (e.g. `MetadataQuery`) which makes it easier to use and reduces errors.
+
+### Seperate packages for Web and Node platforms
+
+  - talk a bit more about this and package specifics 
+  - Introduction of gRPC?
+
+### Improved Functionality
+
+(with examples)
+
+#### InsertMany
+#### Migration?
+#### Client.close()
+#### Filters
+#### Named vectors? 
+#### cleaner return object 
+#### Error handling 
+
+
+## How to migrate your code
+
+The migration will likely involve significant changes to your codebase. Review the [Typescript client library documentation](./index.mdx) to get started, including instantiation details and various submodules.
```

Comment:
This doesn't really say much about migrating code, it's more of a see-also section. Consider renaming it.

It would be useful to lay out a migration path. Maybe something like:

[Update your connection code](link)
[Update queries](link)
[Update inserts and deletes](link)
[Update instance management code](link)

(Or whatever is appropriate) It could incorporate the examples you've suggested above, or link to existing references elsehwere.

Reply:
Updates moved to other branches